includes/bp-members/classes/class-bp-members-endpoints.php

```diff
@@ -0,0 +1,402 @@
+			'type'              => 'integer',
+			'sanitize_callback' => 'absint',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		$params['type'] = array(
+			'description'       => __( 'Limit result set to items with a specific activity type.', 'buddypress' ),
+			'type'              => 'string',
+			'enum'              => ['active', 'newest', 'popular', 'online', 'alphabetical', 'random'],
+			'default'           => 'alphabetical',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		$params['search'] = array(
+			'description'       => __( 'Limit result set to items that match this search query.', 'buddypress' ),
+			'default'           => null,
+			'type'              => 'string',
+			'sanitize_callback' => 'sanitize_text_field',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		$params['meta_key'] = array(
+			'description'       => __( 'Limit result set to items that match this metadata.', 'buddypress' ),
+			'default'           => null,
+			'type'              => 'string',
+			'sanitize_callback' => 'sanitize_text_field',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		$params['meta_value'] = array(
+			'description'       => __( 'Limit result set to items that match this search query.', 'buddypress' ),
+			'default'           => null,
+			'type'              => 'string',
+			'sanitize_callback' => 'sanitize_text_field',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		$params['user_id'] = array(
+			'description'       => __( 'Ensure result set includes specific user IDs.', 'buddypress' ),
+			'type'              => 'array',
+			'default'           => array(),
+			'sanitize_callback' => 'wp_parse_id_list',
+		);
+
+		$params['populate_extras'] = array(
+			'description'       => __( 'Fetch extra meta for each user such as their full name, if they are a friend of the logged in user, their last activity time.', 'buddypress' ),
+			'default'           => true,
+			'type'              => 'boolean',
+			'sanitize_callback' => 'sanitize_text_field',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		return $params;
+	}
+
+	public function prepare_item_for_response( $activity, $request, $is_raw = false ) {
+		$data = array(
+			'author'                => $activity->user_id,
+			'component'             => $activity->component,
+			'content'               => $activity->content,
+			'date'                  => $this->prepare_date_response( $activity->date_recorded ),
+			'id'                    => $activity->id,
+			'link'                  => $activity->primary_link,
+			'parent'                => $activity->type === 'activity_comment' ? $activity->item_id : 0,
+			'prime_association'     => $activity->item_id,
+			'secondary_association' => $activity->secondary_item_id,
+			'status'                => $activity->is_spam ? 'spam' : 'published',
+			'title'                 => $activity->action,
+			'type'                  => $activity->type,
```

Comment:
Sanitize all variables above. Use `intval` for numeric values, `sanitize_key` for lowercase plus hyphen variables (eg `$activity->is_spam`), `wp_kses_post` for content with markup (I will be adding our own at some point to whitelist ONLY the markup we allow, but this is fine for now. Just add a @todo so I can find and change later), `esc_url_raw` for urls, for all others use `sanitize_text_field`.